src/compiler/sys.ts

```diff
@@ -335,7 +335,12 @@ namespace ts {
                     data = "\uFEFF" + data;
                 }
 
-                _fs.writeFileSync(fileName, data, "utf8");
+                if (fileName === "/dev/stdout" || fileName === "-") {
+                    process.stdout.write(data, "utf8");
+                }
+                else {
+                    _fs.writeFileSync(fileName, data, "utf8");

```

Comment:
@mhegazy: _Not_ using `fs.writeFileSync` is at least _currently_ required for supporting output to stdout. With `fs.writeFileSync`, if you target `/dev/stdout`, as previously stated,  you get  error 'ESPIPE, invalid seek' on Linux and 'ENXIO, no such device or address' on OS X. 

> what can we change to make it work for all cases without the special handling?

Talk to the Node.js team to investigate why writing to `/dev/stdout` with `fs.writeFileSync` doesn't work and/or convince them to change / fix it.

Conversely: What problems do you see with this proposed approach to special-casing stdout output?

After all, note that [@MartyIX's quote from the Node.js documentation](https://github.com/Microsoft/TypeScript/pull/5454)  implies that stdout and stderr _are_ special:

 > process.stderr and process.stdout are unlike other streams in Node.js [...]